osidb/models.py

```diff
@@ -696,6 +732,24 @@ def _validate_cvss3(self):
                 "CVSSv3 score is missing",
             )
 
+    def _validate_major_incident_state(self):
+        """
+        Checks that a flaw that is Major Incident has a valid state.
+        Valid states are MI_APPROVED or MI_CISA_APPROVED.
+        """
+        invalid_states = [
+            self.FlawMajorIncident.MI_UNKNOWN,
+            self.FlawMajorIncident.MI_REQUESTED,
+            self.FlawMajorIncident.MI_REJECTED,
+            self.FlawMajorIncident.MI_INVALID,
+        ]
+
+        if self.is_major_incident and self.major_incident_state in invalid_states:
+            self.alert(
+                "mi_state_invalid",
+                "A flaw marked as Major Incident does not have a valid state.",
+            )
```

Comment:
I think that the correct validation would be simply like

`if self.major_incident_state == self.FlawMajorIncident.MI_INVALID: ...alert`